### ISSUE TITLE
Rollup & Angular Material is broken since angular 4.2

### ISSUE THREAD

**Issue by Karasuni** [mgechev/angular-seed#1970]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see use [gitter](https://gitter.im/mgechev/angular2-seed) or [stackoverflow](https://stackoverflow.com/questions/tagged/angular2)
```

**Current behavior**
When you add angular-material to the project it is not able to build since the upgrade of angular 4.1 -> 4.2.

**Minimal reproduction of the problem with instructions**

1. Integrate angular-material ([minimal example](https://github.com/Karasuni/angular-seed/tree/material))
2. `npm i` and `npm run build.prod.rollup.aot`

You will receive the error:
```
{ Error: A module cannot import itself
    at error (...\angular-seed\node_modules\rollup\src\utils\error.js:2:14)
    at Module.error$1 [as error] (...\angular-seed\node_modules\rollup\src\Module.js:303:2)
    at ...\angular-seed\node_modules\rollup\src\Bundle.js:399:15
  code: 'CANNOT_IMPORT_SELF',
  pos: 166,
  loc: 
   { file: '...\\angular-seed\\dist\\tmp\\node_modules\\@angular\\material\\typings\\index.ngfactory.js',
     line: 5,
     column: 0 },
  frame: '3: import * as i2 from \'@angular/common\';\n4: import * as i3 from \'@angular/platform-browser\';\n5: import * as i4 from \'./index.ngfactory\';\n   ^\n6: import * as i5 from \'@angular/forms\';\n7: import * as i6 from \'@angular/http\';' }
```

**Affected version:** 71329709dc5fa463e6152908b9e9ca2f5ee72505

I'm not sure if the issue is sourced at `angular` or at `angular/material`. The issue was reported also at https://github.com/angular/angular/issues/17389 and https://github.com/steveblue/angular2-rollup/issues/18 but there is no solution yet.

I've compared with an earlier commit using angular 4.1 and [this version](https://github.com/Karasuni/angular-seed/tree/material-4.1) builds flawlessly.

**Comment by cracrayol** [comment]
It was an angular problem fixed in version 4.2.2 (see https://github.com/angular/angular/commit/ed73d4f3ac6b542bf5ea3eb73fbe91e2ceabcdb4)

**Comment by vyakymenko** [comment]
@Karasuni , to resolve this issue for some time until angular community is working on this you can make this simple steps:

1. Delete `npm_modules` directory.
2. Open `package.json` file via any editor.
2. Change dependencies to this one:

```json
  ...
  "dependencies": {
    "@angular/animations": "4.1.3",
    "@angular/common": "4.1.3",
    "@angular/compiler": "4.1.3",
    "@angular/core": "4.1.3",
    "@angular/forms": "4.1.3",
    "@angular/http": "4.1.3",
    "@angular/platform-browser": "4.1.3",
    "@angular/platform-browser-dynamic": "4.1.3",
    "@angular/router": "4.1.3",
    "@angular/service-worker": "^1.0.0-beta.15",
    .....
  }
}
```

4. In terminal type `$ npm install`.

Then everything will work.

**Comment by vyakymenko** [comment]
@Karasuni , it's an `angular-compiler` issue. So for now best and fast way to resolve the issue, revert and re-install this packages:

`package.json`

```json
"devDependencies": {
    "@angular/compiler-cli": "~4.1.3",
    "@angular/platform-server": "~4.1.3",
   ....
  },
  "dependencies": {
    "@angular/animations": "~4.1.3",
    "@angular/common": "~4.1.3",
    "@angular/compiler": "~4.1.3",
    "@angular/core": "~4.1.3",
    "@angular/flex-layout": "^2.0.0-beta.8",
    "@angular/forms": "~4.1.3",
    "@angular/http": "~4.1.3",
    "@angular/material": "^2.0.0-beta.6",
    "@angular/platform-browser": "~4.1.3",
    "@angular/platform-browser-dynamic": "~4.1.3",
    "@angular/router": "~4.1.3",
    "@angular/service-worker": "^1.0.0-beta.15",
    ...
  },
```
Remove `node_modules` directory, and re-install `$ npm install`.

For `Node 8` users don't forget before installing, delete `package.json.lock`.

**Comment by patrickmichalina** [comment]
This should probably be close since it isn't a Seed issue

**Comment by amalshehu** [comment]
I rolled back to 4.1.3. But still has the issue.Any workarounds? @vyakymenko

**Comment by Karasuni** [comment]
Try deleting node_modules and running `npm install` again.

**Comment by vyakymenko** [comment]
@amalshehu , you need to make few steps:

1. In your project delete `node_modules` directory.
2. Open your `package.json` file and look for all dependencies that I noticed before.
```
"devDependencies": {
    "@angular/compiler-cli": "~4.1.3",
    "@angular/platform-server": "~4.1.3",
   ....
  },
  "dependencies": {
    "@angular/animations": "~4.1.3",
    "@angular/common": "~4.1.3",
    "@angular/compiler": "~4.1.3",
    "@angular/core": "~4.1.3",
    "@angular/flex-layout": "^2.0.0-beta.8",
    "@angular/forms": "~4.1.3",
    "@angular/http": "~4.1.3",
    "@angular/material": "^2.0.0-beta.6",
    "@angular/platform-browser": "~4.1.3",
    "@angular/platform-browser-dynamic": "~4.1.3",
    "@angular/router": "~4.1.3",
    "@angular/service-worker": "^1.0.0-beta.15",
    ...
  },
```
3. Change them.
4. Type `$ npm install`.

**Comment by amalshehu** [comment]
Dude i reverted to 4.0.0, It started working.You're correct its an compiler issue.Waiting for a fix from angular  team on next 4.2.2 release. @vyakymenko
